src/spec/payid-easy-checkout.md

```diff
@@ -0,0 +1,380 @@
+                "template": "https://wallet.com/checkout?amount={amount}&receiverPayId={receiverPayId}&currency={currency}&nextUrl={nextUrl}"
+            }
+        ]
+     }
+     
+The receiver client must parse this response to find the PayID Easy Checkout Link. 
+If the JRD returned from the PayID Easy Checkout Discovery query does not contain a 
+PayID Easy Checkout Link in its 'links' collection, PayID Easy Checkout is considered to have failed.
+Once a PayID Easy Checkout URI Template has been obtained from the PayID Easy Checkout Link by the PayID client, 
+PayID Easy Checkout Discovery is considered to be complete. 
+
+### Template Syntax
+This specification defines a simple template syntax for PayID Easy Checkout URI Template
+transformation.  A template is a string containing brace-enclosed
+("{}") variable names marking the parts of the string that are to be
+substituted by the corresponding variable values.
+
+This specification defines several variables, which MAY or MAY NOT be present in every PayID Easy Checkout URI Template.
+These variables are as follows:
+    
+    'amount': The amount that should be sent by the sender to the receiver
+    'receiverPayID': The PayID URI of the receiver
+    'currency': The ISO-4217 currency code that the sender should send
+    'network': The network that the sender should send payment over (TODO: define network enum or use rfc)
+    'nextURL': A URL that the sender's wallet can use after completing the payment
+    
+When substituting values into a URI 'path' or 'query' part as defined by
+[RFC3986][], values with characters outside the character set allowed by paths or query parameters in [RFC3986][], 
+respectively, MUST be percent or otherwise encoded.
+
+Protocols MAY define additional variables and syntax rules, but MUST NOT
+change the meaning of the variables specified in this document. If a client is unable to
+successfully process a template (e.g., unknown variable names, unknown or
+incompatible syntax), the link SHOULD be ignored.
+
+The template syntax ABNF is as follows:
+
+    uri-char     =  ( reserved / unreserved / pct-encoded )
+    var-char     =  ALPHA / DIGIT / "." / "_"
+    var-name     =  %x61.63.63.74.70.61.72.74 / ( 1*var-char )
+    variable     =  "{" var-name "}"
+    PAYID-EASY-CHECKOUT-URI-Template =  *( uri-char / variable )
+
+For example:
+
+    Input:    alice$wallet.com
+              amount = 10
+              receiverPayID=pay$merchant.com
+              currency=XRP
+              network=XRPL
+              nextUrl=https://merchant.com/thankyou
+    Template: https://wallet.com/checkout?amount={amount}&receiverPayId={receiverPayID}&currency={currency}&network={network}&nextUrl={nextURL}
+    Output:   https://wallet.com/checkout?amount=10&receiverPayId=payid%2Apay%24merchant.com&currency=XRP&network=XRPL&nextUrl=https://merchant.com/thankyou
+
+TODO: Should we define acceptable URL template variable values for the redirect?
```

Comment:
Open question here.  I believe in the charity demo we redirect back to the charity site from the wallet with some populated data.  I could imagine a merchant wanting to get back a transaction id or payment status or something.  If so, we would need to spec out a URI template format for the `nextUrl` field in the PayID Easy Checkout URI Template.

My thinking here though is that this is probably not needed for now.  Merchants/payment processors can likely just show a generic confirmation page (maybe with an invoiceId or something, which can be put in `nextUrl`), and that they won't need any information about the payment in the redirect URL, because they will already have systems in place to correlate payments and monitor ledgers.

Thoughts on this?

Reply:
Per our conversation, no spec needed, but we'll add an example in section 8 that includes a `transactionId` in the PayID (e.g., `pay-123456789$example.com`).